diceRoll/diceRollUnitTests.js

```diff
@@ -65,63 +38,118 @@ function assert(condition, message) {
   }
 }
 
+function stub() {
+  var info = {
+    called: {
+      true: false,
+      expectedCount: null,
+      count: 0,
+      check: function(expectedCount) {
+        info.called.expectedCount = expectedCount;
+        if (info.called.expectedCount === undefined || info.called.expectedCount === null) {
+          return info.called.true;
+        } else {
+          return (info.called.count === info.called.expectedCount);
+        }
+      }
+    },
+    args: {
+      expected: null,
+      current: null,
+      all: {
+        array: [],
+        check: function(index, expectedArray) {
+          var index = arguments[0];
+          info.args.expected = {};
+          for (key in arguments) {
+            if (key !== "0") {
+              info.args.expected[key - 1] = arguments[key];
+            }
+          }
+          return (JSON.stringify(info.args.expected) === JSON.stringify(info.args.all.array[index]));
+        }
+      },
+      log: function(current) {
+        info.args.all.array[info.called.count] = current;
+      },
+      check: function(expectedArgs) {
+        info.args.expected = arguments;
```

Comment:
Do you need to store expected? If you don't use it outside of this function then don't add it to info.